runtime/interpreter/value.go

```diff
@@ -785,6 +827,45 @@ func (v *ArrayValue) Count() int {
 	return len(v.Values)
 }
 
+func (v *ArrayValue) ConformsToDynamicType(dynamicType DynamicType) bool {
+	arrayType, ok := dynamicType.(ArrayDynamicType)
+	if !ok || len(v.Values) != len(arrayType.ElementTypes) {
+		return false
+	}
+
+	for index, item := range v.Values {
+		if !item.ConformsToDynamicType(arrayType.ElementTypes[index]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (v *ArrayValue) conformsToSemaType(semaType sema.Type) bool {
+	var memberType sema.Type
+
+	switch arrayType := semaType.(type) {
+	case *sema.VariableSizedType:
+		memberType = arrayType.Type
+	case *sema.ConstantSizedType:
+		if int64(len(v.Values)) != arrayType.Size {
+			return false
+		}
+		memberType = arrayType.Type
+	default:
+		return false
+	}
+
+	for _, member := range v.Values {
+		if !ValueConformsToSemaType(member, memberType) {
+			return false
```

Comment:
Minor: I was confused about "members" in arrays at first (we normally only use this term to refer to fields and functions in composites), so maybe use "element" for arrays here:

```suggestion
	var elementType sema.Type

	switch arrayType := semaType.(type) {
	case *sema.VariableSizedType:
		elementType = arrayType.Type
	case *sema.ConstantSizedType:
		if int64(len(v.Values)) != arrayType.Size {
			return false
		}
		elementType = arrayType.Type
	default:
		return false
	}

	for _, element := range v.Values {
		if !ValueConformsToSemaType(element, elementType) {
			return false
```

Reply:
Added in 4fd3be9